### ISSUE TITLE
Segfault when running manimgl

### ISSUE THREAD

**Issue by eulertour** [3b1b/manim#1387]
### Describe the bug
Running `manimgl` with or without arguments results in a segfault

**Code**:
n/a

**Wrong display or Error traceback**:
![printscreen](https://user-images.githubusercontent.com/43117506/107866198-e7183980-6e22-11eb-8afc-452a65efc4fc.png)

### Additional context
The crash began after #1383 was merged.

**Comment by naveen521kk** [comment]
@eulertour Can you run with the debugger and post the backtrace?
```sh
gdb -batch -ex "run" -ex "bt" manimgl
```

**Comment by naveen521kk** [comment]
Also, if the problem is with manimpango, can you check whether it works with ManimCE?

**Comment by TonyCrane** [comment]
I went back to the version you mentioned, and then created a new conda environment, installed manimgl and ran it without any errors.
```sh
git checkout ad332c586341a665b2e69d4dbea8b023c66d575d
conda create -n manim_1387 python=3.8
source activate manim_1387
pip install -e . -i https://pypi.tuna.tsinghua.edu.cn/simple/
manimgl
```
And I feel that there is nothing in #1383 that will cause a segfault

**Comment by eulertour** [comment]
@naveen521kk The first command
![printscreen](https://user-images.githubusercontent.com/43117506/107867767-5b0e0e00-6e32-11eb-8dba-896a1af2c542.png)

the second command
![printscreen](https://user-images.githubusercontent.com/43117506/107867779-7da02700-6e32-11eb-8b7f-d3521dec248c.png)

it works with ManimCE.

@TonyCrane If you weren't aware, ManimPango is a custom-written C extension for using Pango.
https://github.com/ManimCommunity/ManimPango/blob/main/manimpango/cmanimpango.pxd

**Comment by naveen521kk** [comment]
does this command show Seg Fault?
```sh
python -c "import manimpango"
```

**Comment by eulertour** [comment]
No

**Comment by naveen521kk** [comment]
for the results about using gdb, you don't have debugging symbols installed see https://wiki.python.org/moin/DebuggingWithGdb

**Comment by eulertour** [comment]
You don't have to help, I almost never use `Text` anyway.

I'm suggesting to make the change optional because the functionality isn't necessary to use manim and the code is unstable.

**Comment by naveen521kk** [comment]
rather try running python using the verbose mode, 
```sh
python -v -m manimlib
```

**Comment by naveen521kk** [comment]
> I'm suggesting to make the change optional because the functionality isn't necessary to use manim and the code is unstable.

Try removing `text_mobject.py` and run it also fix imports? I kinda feel like this isn't because of ManimPango.

**Comment by eulertour** [comment]
I made a pastebin here: https://pastebin.com/dzV8dJpW.
manim runs normally if I comment out the imports of `manimlib.mobject.svg.text_mobject` and `manimlib.animation.numbers` (which imports text_mobject).

**Comment by naveen521kk** [comment]
The paste says, https://pastebin.com/dzV8dJpW, the import `manimlib.mobject.svg.text_mobject` was successful which means there is nothing to do with ManimPango. line 1264

so after the `import 'pyglet.gl' # <_frozen_importlib_external.SourceFileLoader object at 0x7fd22d9faf70>` it shows Seg Fault? Try checking out to https://github.com/3b1b/manim/commit/96d9a66faf59bc97f709ebdd8a10685e1e141da5 and see if there is a Seg Fault there?

I think this should be the cause of Seg Faults https://github.com/3b1b/manim/commit/368a6c3d1ca8dadc5e2808d551a3746f2b192c40

**Comment by naveen521kk** [comment]
@eulertour try setting env var `PYTHONFAULTHANDLER` to 1. Send what it says.  see https://docs.python.org/3/library/faulthandler.html

**Comment by eulertour** [comment]
Running on both 96d9a66 and 368a6c3 runs normally. Dumping the traceback gives https://pastebin.com/HqsELR16.

I don't know if the problem is caused by ManimPango either. From what I've looked at it seems to be some interaction between ManimPango and moderngl that causes the problem. But being able to use manim is more important than being able to use pango, so to prevent breaking the entire library it makes more sense just to make it optional.

**Comment by naveen521kk** [comment]
Just a last try, remove this line in `__init__.py` and run it?
https://github.com/3b1b/manim/blob/b7e2abb8016021f468c2f4c98790c94a75040a3f/manimlib/__init__.py#L26

**Comment by eulertour** [comment]
It still segfaults when that line is commented, and I verified that the import wasn't occurring anywhere else.

**Comment by naveen521kk** [comment]
Try compiling manimpango yourselves, maybe that fixes it?
```sh
python -m pip install manimpango --no-binary :all: --force
```

**Comment by eulertour** [comment]
Compiling locally fixes the problem on ad332c5 and master.

**Comment by naveen521kk** [comment]
Ah, I understand what happens, there are two different Pango version one used by manimpango and the other one from your system. Since, it's linux and gnome, any window which is opened uses pango for text rendering and this conflicting version, causing a segmentation fault. That's a reason I always recommend Linux users to install without using wheels, https://github.com/manimcommunity/manimpango#note, and I created those wheels only for the case of RTD build on Community Edition. It's also a notable reason why there don't exist wheels for pycairo outside of Windows.

For other OS like Mac and windows, this doesn't happen because the system uses its own rendering system and not Pango.
